src/rotation_types.jl

```diff
@@ -0,0 +1,192 @@
+
+# default parameters
+default_params{T <: Quaternion}(::Type{T}) = (@DefaultElType(), ) 
+
+# Quaternions have an extra bool field that type_methods.jl doesn't want to know about
+numel(::Type{Quaternion}) = 4 
+
+
+
+###################################################
+# Stereographic projection of Quaternions
+###################################################
+
+@doc """
+
+A 3 element parametrization of a unit quaternion Q formed by the intersection of a line from [-1,0,0,0] to Q, with a plane containing the origin and with normal direction [1,0,0,0]. This
+is a compact representation of rotations where the derivitives of the rotation matrix's elements wrt the SpQuat parameters are rational functions (good for optimization)
+
+See:
+
+    [1] "A Recipe on the Parameterization of Rotation Matrices for Non-Linear Optimization using Quaternions",
+    Terzakis, Culverhouse, Bugmann, Sharma, Sutton, 
+    MIDAS technical report, 2011
+    http://www.tech.plymouth.ac.uk/sme/springerusv/2011/publications_files/Terzakis%20et%20al%202012,%20A%20Recipe%20on%20the%20Parameterization%20of%20Rotation%20Matrices...MIDAS.SME.2012.TR.004.pdf
+
+    Note 1: the singularity (origin) has been moved from [0,0,0,-1] in Ref[1] to [-1,0,0,0], so the 0 rotation quaternion [1,0,0,0] maps to [0,0,0] as opposed of to [1,0,0]. 
+    Note 2: the quaternion rotation ambiguity q = -q means there will be a rotation with ||SpQuat|| <= 1 and an equivilent rotation with ||SpQuat|| > 1.  This package
+            uses the solution with ||SpQuat|| <= 1 
+
+""" ->
+immutable SpQuat{T <: Real}
+    x::T
+    y::T
+    z::T
+end
+
+
+# convert to a known one
+convert_rotation{T <: Real}(::Type{Quaternion{T}}, spq::SpQuat{T}) = spquat_to_quat(spq)
+convert_rotation{T <: Real}(::Type{SpQuat{T}}, q::Quaternion{T}) = quat_to_spquat(q)
+
+# define its interaction with other angle representations
+convert_rotation{T <: Real}(::Type{SpQuat{T}}, R::RotMatrix{T}) = quat_to_spquat(convert_rotation(Quaternion{T}, R))
+convert_rotation{T,U}(::Type{T}, spq::SpQuat{U}) = convert_rotation(T, spquat_to_quat(promote_eltype(spq, T))) # this has template overload problems without the U????
+
+# default parameters
+default_params{T <: SpQuat}(::Type{T}) = (@DefaultElType(), )
+
+# define an inverse for the SPQuat since its trivial.  Inverse in the sense of the corresponding inverse rotation 
+inv(X::SpQuat) = SpQuat(-X.x, -X.y, -X.z) 
+
+# define multiplication for SpQuat corresponding to combining rotations
+*(lhs::SpQuat, rhs::SpQuat) = SpQuat(Quaternion(lhs) * Quaternion(rhs))
+
+# pull the rotation angle as well
+rot_angle(X::SpQuat) = rot_angle(Quaternion(X))
+
+
+
+###################################################
+# Arbitrary axis rotation representations
+# (limited support for now)
+###################################################
+
+
+immutable AngleAxis{T <: Real}  
```

Comment:
I see angle axis as just a human understandable parametrization so I'd argue for the above being a different type.  You also couldn't represent a rotation about a fixed axis by theta < 0